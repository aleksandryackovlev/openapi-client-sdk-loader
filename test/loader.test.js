import { getRunner, getRunnerResult } from './helpers';

describe('loader', () => {
  it('should work', async () => {
    const runner = getRunner('petstore.yaml');
    const stats = await runner();

    expect(getRunnerResult(stats)).toMatchSnapshot('runner_result');
  });
});
