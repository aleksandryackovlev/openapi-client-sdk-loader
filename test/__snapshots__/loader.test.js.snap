// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should work: runner_result 1`] = `
"export interface Category {
  id?: number;
  name?: string;
}

export interface Pet {
  id?: number;
  category?: Category;
  name: string;
  photoUrls: string[];
  tags?: Tag[];
  /**
   * pet status in the store
   */
  status?: \\"available\\" | \\"pending\\" | \\"sold\\";
}

export interface Tag {
  id?: number;
  name?: string;
}

export interface ApiResponse {
  code?: number;
  type?: string;
  message?: string;
}

export interface Order {
  id?: number;
  petId?: number;
  quantity?: number;
  shipDate?: string;
  /**
   * Order Status
   */
  status?: \\"placed\\" | \\"approved\\" | \\"delivered\\";
  complete?: boolean;
}

export interface User {
  id?: number;
  username?: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  password?: string;
  phone?: string;
  /**
   * User Status
   */
  userStatus?: number;
}

export type OffsetParamParam = number;

export interface AnotherParamParam {
  dateFrom?: string;
  dateTo?: string;
}

export type ErrorRequestBodyJson = User;

export interface ErrorRequestBodyXml {
  id?: string;
  title?: string;
}

const runtime = \\"runtime line\\";
const apis = \\"apis line\\";
"
`;
